### PR TITLE
21737 - Create alembic script & model for furnishings table

### PR DESCRIPTION
**Pull request by chenhongjing**
*Issue #:* /bcgov/entity#21737

*Description of changes:*
- Added alembic script to create furnishings table
- Added data model for furnishing
- Added unit tests
- Misc updates for others

*db after upgrading*
image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/migrations/versions/01b28a2bb730_furnishings.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            'INTENT_TO_DISSOLVE',
+            'INTENT_TO_DISSOLVE_XPRO',
+            'CORP_DISSOLVED',
+            'CORP_DISSOLVED_XPRO',
+            name='furnishing_name'
+            ),
+            nullable=False
+        ),
+        sa.Column('batch_id', sa.Integer(), nullable=False),
+        sa.Column('grouping_identifier', sa.Integer(), nullable=True),
```
Comment (argush3) [inline legal-api/migrations/versions/01b28a2bb730_furnishings.py:38]: Can we add a sequence that we can use for this field when it's required?

Thinking something like this.  Can start at 1.

https://github.com/bcgov/lear/blob/c8572bcc07bcbe8c1be41b54dd080463a43ec96d/legal-api/migrations/versions/3d8eb786f4c2_add_coop_sequence.py#L19-L24

File: legal-api/src/legal_api/models/address.py [hunk 2]
```diff
@@ -32,7 +32,8 @@ class Address(db.Model):  # pylint: disable=too-many-instance-attributes
 
     MAILING = 'mailing'
     DELIVERY = 'delivery'
-    ADDRESS_TYPES = [MAILING, DELIVERY]
+    FURNISHING = 'furnishing'
+    ADDRESS_TYPES = [MAILING, DELIVERY, FURNISHING]
```
Comment (argush3) [inline legal-api/src/legal_api/models/address.py:36]: I know i have this in the design but I'm actually still thinking about how we will model the address used at the time the dissolution letters need to go out.

Can we leave this out for now?

**Comment by argush3** [comment]
Looks good.  Just the few comments.

**Review by leodube-aot — APPROVED** [review]
LGTM!

**Review by argush3 — APPROVED** [review]

**Pull request merged**